### PR TITLE
Scaffold Next.js 16 project with game dependency set

### LINKED ISSUE #3: Scaffold Next.js 16 project with game dependency set
## Objective
Initialize the greenfield Deadbolt project with Next.js 16 (App Router), TypeScript, Tailwind CSS, and all game dependencies. Establish the directory structure that separates the React application shell from the pure-TypeScript game engine.

## Acceptance Criteria
- [ ] Next.js 16 project created with App Router, TypeScript, Tailwind CSS, ESLint, and `src/` directory convention using pnpm
- [ ] All game dependencies installed: phaser, zustand, miniplex, pathfinding, seedrandom, eventemitter3 (or similar typed event emitter)
- [ ] shadcn/ui initialized with base components: Button, Card, Dialog, Sheet, Progress, Tooltip
- [ ] Directory skeleton created: `src/game/`, `src/game/ecs/`, `src/game/scenes/`, `src/game/systems/`, `src/game/procgen/`, `src/stores/`, `src/lib/`, `src/types/`, `public/assets/sprites/`, `public/assets/tilemaps/`, `public/assets/audio/sfx/`, `public/assets/audio/music/`
- [ ] Geist Sans and Geist Mono fonts configured via `next/font`
- [ ] Dark mode set as default on the root `<html>` element
- [ ] TooltipProvider wrapped at layout root per shadcn/ui convention
- [ ] `pnpm build` completes successfully with zero errors
- [ ] CLAUDE.md created with project conventions (stack, directory layout, bridge pattern, ECS-only game logic rule)

## Technical Notes
- Use `pnpm create next-app@16` with `--typescript --tailwind --app --src-dir --eslint` flags
- The `src/game/` directory must contain zero React imports — this boundary is architecturally critical
- Path alias `@/*` should map to `./src/*` in tsconfig
- Turbopack is the default bundler in Next.js 16 — verify Phaser loads correctly with it

## Dependencies
None — this is the first issue in the project.

### PR DESCRIPTION
**Pull request by allan-mobley-jr**
## Summary
Closes #3: Initialize the Deadbolt project with Next.js 16, TypeScript, Tailwind CSS, game dependencies, shadcn/ui components, and a complete directory skeleton for the game engine.

## Changes
- Next.js 16 (App Router, Turbopack, React 19) with TypeScript strict mode
- Game dependencies: Phaser 3, Miniplex 2, Zustand 5, PathFinding.js, seedrandom, eventemitter3
- shadcn/ui (base-nova) components: Button, Card, Dialog, Sheet, Progress, Tooltip
- Directory skeleton: `src/game/{ecs,scenes,systems,procgen}`, `src/stores`, `src/types`, `public/assets/*`
- Geist Sans + Geist Mono fonts via next/font, dark mode default, TooltipProvider at layout root
- CLAUDE.md with project conventions, architecture rules, and quality gates
- **Vitest test infrastructure** with 31 tests across 3 suites
- **GitHub Actions CI workflow** (Quality Checks: lint, typecheck, test, build)

## Test Coverage
- `src/lib/utils.test.ts` — 6 tests for the `cn()` utility (class merging, deduplication, conditionals, edge cases)
- `src/app/page.test.tsx` — 1 render smoke test (heading + subtitle)
- `tests/scaffold.test.ts` — 24 structural assertions (10 directories, 6 shadcn components, 2 project files, 6 game dependencies)

## Acceptance Criteria
- [x] Next.js 16 project created with App Router, TypeScript, Tailwind CSS, ESLint, and `src/` directory convention using pnpm
- [x] All game dependencies installed: phaser, zustand, miniplex, pathfinding, seedrandom, eventemitter3
- [x] shadcn/ui initialized with base components: Button, Card, Dialog, Sheet, Progress, Tooltip
- [x] Directory skeleton created: `src/game/`, `src/game/ecs/`, `src/game/scenes/`, `src/game/systems/`, `src/game/procgen/`, `src/stores/`, `src/lib/`, `src/types/`, `public/assets/sprites/`, `public/assets/tilemaps/`, `public/assets/audio/sfx/`, `public/assets/audio/music/`
- [x] Geist Sans and Geist Mono fonts configured via `next/font`
- [x] Dark mode set as default on the root `<html>` element
- [x] TooltipProvider wrapped at layout root per shadcn/ui convention
- [x] `pnpm build` completes successfully with zero errors
- [x] CLAUDE.md created with project conventions (stack, directory layout, bridge pattern, ECS-only game logic rule)

## Quality Checks
- [x] Lint passes
- [x] Type check passes
- [x] Tests pass (31 total)
- [x] Build succeeds

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+yarn dev
+# or
+pnpm dev
+# or
+bun dev
+```
+
+Open [http://localhost:3000](http://localhost:3000) with your browser to see the result.
+
+You can start editing the page by modifying `app/page.tsx`. The page auto-updates as you edit the file.
```
[the remark this answers is not in the document]
Comment (allan-mobley-jr) [reply README.md]: Fixed in 4c7f124. Corrected path to `src/app/page.tsx` to match the `src/` directory convention.

File: src/app/page.test.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import { render, screen, cleanup } from '@testing-library/react'
+import Page from '@/app/page'
+
+test('Home page renders heading and subtitle', () => {
+  render(<Page />)
+  expect(
+    screen.getByRole('heading', { level: 1, name: 'Deadbolt' })
+  ).toBeDefined()
+  expect(screen.getByText('Zombie survival base builder')).toBeDefined()
+  cleanup()
```
[the remark this answers is not in the document]
Comment (allan-mobley-jr) [reply src/app/page.test.tsx]: Fixed in 4c7f124. Moved cleanup to `afterEach(cleanup)` so it runs even if assertions throw. Also upgraded from `toBeDefined()` to `toBeInTheDocument()` via `@testing-library/jest-dom` for stronger guarantees.
